mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -105,6 +105,15 @@ bestVectorization(OpBuilder &b, Value matrix, int64_t dataPerThread,
       transforms, static_cast<uint32_t>(GemmDimension::MorN),
       math_util::gcd(dataPerThread, dPerBlock), tensorShape);
 
+  // Vectorizing more than the physical vector length (128 bits) might
+  // be harmful for coalescence and other metrics. Let's limit the maximum
+  // amount of data to load to the maximum vector length. This means a
+  // warp will issue, if possible, a global_load_dwordx4 instruction
+  const int64_t maxVectorLenBits = 128;
+  auto bwidth = elementType.getIntOrFloatBitWidth();
+  kVectorLen = std::min(maxVectorLenBits / bwidth, kVectorLen);
```

Comment:
One note: gcd, not min. A best vectorization of N means that you can vectorize with any `d` that divides `N`, not any `d < N`

Reply:
Yes, you are right, good catch!